pkg/tests/generic.go

```diff
@@ -0,0 +1,51 @@
+package tests
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/nimrodshn/cs-load-test/pkg/helpers"
+	"github.com/nimrodshn/cs-load-test/pkg/result"
+	vegeta "github.com/tsenart/vegeta/v12/lib"
+)
+
+func TestGenericEndpoint(options *helpers.TestOptions) error {
+
+	testName := options.TestName
+	log.Printf("Executing Test: %s", testName)
+	log.Printf("Rate: %s", options.Rate.String())
+	log.Printf("Duration: %s", options.Duration.String())
+	log.Printf("Path: %s", options.Path)
+
+	// Vegeta Results File
+	fileName := fmt.Sprintf("%s_%s.json", options.ID, testName)
+	resultFile, err := createFile(fileName, options.OutputDirectory)
+	defer resultFile.Close()
+	if err != nil {
+		return err
+	}
+
+	// Specify the HTTP request(s) that will be executed
+	target := vegeta.Target{
+		Method: options.Method,
+		URL:    options.Path,
+	}
+	if len(options.Body) > 0 {
+		target.Body = []byte(options.Body)
+	}
+	targeter := vegeta.NewStaticTargeter(target)
```

Comment:
I'm afraid this won't work :cry:  b/c some tests (specifically `CreateCluster`) require generating dynamic `Targerter` at each attack otherwise `AMS` will block creating clusters with the same name (After the first req). I suspect `RegisterCluster` will also fail if you try and send the same cluster more than once.

One way to resolve this would be to pass a `Targeter` as part of `TestOptions`; I will give it some more thought..

I like the direction though.

Reply:
Yeah, I agree that this won't work for CreateCluster and other more complicated tests. This test handler was specifically written for the simple case of "just send a simple request to a given endpoint". Other tests, like CreateCluster, will have their own handler.